sklearn/linear_model/_base.py

```diff
@@ -518,7 +519,38 @@ def fit(self, X, y, sample_weight=None):
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
-        if sp.issparse(X):
+        if cholesky:
+            n_samples, n_features = X.shape
```

Comment:
Is this logic copied from Ridge, and if so, can it be refactored?

Reply:
Yes - newest push refactors this into `_ridge_solvers.py`